test/integration/01b-create-users.spec.ts

```diff
@@ -132,9 +134,34 @@ describe("username tests", () => {
     await expect(userWallet2.setUsername({ username })).rejects.toThrow()
   })
 
-  // FIXME: failing for some reason
-  // it('sets account status correctly', async () => {
-  //   await setAccountStatus({uid: userWallet2._id, status: 'locked'})
-  //   await expect(userWallet2.status).toBe('locked')
-  // })
+  it("sets account status correctly", async () => {
+    await setAccountStatus({ uid: userWallet2.user._id, status: "locked" })
+    userWallet2 = await getUserWallet(2)
+    await expect(userWallet2.user.status).toBe("locked")
+  })
+
+  it("save 2fa for user0", async () => {
+    const { secret } = userWallet0.generate2fa()
+    const token = generateTokenHelper({ secret })
+    await userWallet0.save2fa({ secret, token })
+    userWallet0 = await getUserWallet(0)
+    expect(userWallet0.user.twoFactorEnabled).toBe(true)
+    expect(userWallet0.user.twoFactor.secret).toBe(secret)
+  })
+
+  it("validate 2fa for user0", async () => {
+    const secret = userWallet0.user.twoFactor.secret
+    const token = generateTokenHelper({ secret })
+    expect(
+      UserWallet.throwIf2faNotValid({ token, logger: baseLogger, secret }),
+    ).toBeTruthy()
+  })
+
+  it("delete 2fa for user0", async () => {
+    const token = generateTokenHelper({ secret: userWallet0.user.twoFactor.secret })
+    const result = await userWallet0.delete2fa({ token })
+    expect(result).toBeTruthy()
+    userWallet0 = await getUserWallet(0)
+    expect(userWallet0.user.twoFactorEnabled).toBeFalsy()
+  })
 })
```

Comment:
Is there a reason the `userWallet0` variable is re-used across many scopes?  Wouldn't this introduce test order dependency?

Reply:
Yep, that's how we're doing our tests at the moment. They're all ordered and you can't run most of them independently.
Just to be clear, this is not introducing any new dependency that's not already there. I think removing the ordering dependency will be part of a much bigger refactor.